src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs

```diff
@@ -1347,16 +1284,15 @@ protected override void ProcessRecord()
                 bool matched;
                 MatchInfo result;
                 MatchInfo matchInfo = null;
-                var line = _inputObject.BaseObject as string;
-                if (line != null)
+                if (_inputObject.BaseObject is string line)
```

Comment:
Since we have already refactored a lot I'd use here switch with string, MatchInfo and default cases.